src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -34,31 +41,74 @@
  * name but different signatures where the signature can differ by the number of
  * input parameters or type of input parameters or both.
  * </p>
+ * <ul>
+ * <li>
+ * Property {@code modifierGroups} - control how different types of overloaded methods
+ * can be grouped by their modifiers.
+ * Type is {@code java.util.regex.Pattern[]}.
+ * Default value is {@code ""}.
+ * </li>
+ * </ul>
  * <p>
  * To configure the check:
  * </p>
  * <pre>
- * &lt;module name=&quot;OverloadMethodsDeclarationOrder&quot;/&gt;
+ * &lt;module name="OverloadMethodsDeclarationOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example of incorrect grouping of overloaded methods:
+ * </p>
+ * <pre>
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void notFoo() {} // OK
+ * public void foo(String s, int i) {} // Violation, should be after 'foo(int i, String s)'
```

Comment:
Can we use 1 set of code between all "bad" examples. It will show default behavior, what its like with modifiers not set, and then show the difference when modifiers is set. It will show evolution and configuration that can be done with the different options. Each example will obviously show different order of "good" examples based on configuration.